src/racer/typeinf.rs

```diff
@@ -334,6 +334,15 @@ pub fn get_return_type_of_function(fnmatch: &Match, contextm: &Match, session: &
         ast::parse_fn_output(decl, Scope::from_match(fnmatch))
     });
 
+    // Convert output arg of type Self to the correct type
+    if let Some(core::Ty::PathSearch(ref mut path, _)) = out {
+        if let Some(ref mut path_seg) = path.clone().segments.get(0) {
```

Comment:
I'm still not that good at rust so I copied the code from some lines below.
but do now not know why I have "ref mut" either 
is this better?
`    if let Some(core::Ty::PathSearch(ref path, _)) = out {
        if let Some(ref path_seg) = path.segments.get(0) {
            if "Self" == path_seg.name {
`
I can remove the ref in Some with all test passing also but do not know what is best when I tried to remove the ref in the first stamtent the borrow checker complained.